### DISCUSSION TITLE
FM and DAB+ in parallel possible?

### DISCUSSION THREAD

**Discussion by Apfelsafft** [AlbrechtL/welle.io#839] [Q&A] [answered]
Hello,

besides streaming the DAB+ radio station with welle-cli I'd love to also stream FM stations.
Somewhere in an old issue I read about rtl_fm but I am not sure if it runs in parallel with welle-cli using the same USB device.

Maybe someone can help on that (or has some links for further readings. the rtl-sdr page is quite overloaded with info if you start with it)

happy x-mas
René

**Comment by andimik** [comment]
> I am not sure if it runs in parallel with welle-cli using the same USB device.

Technically impossible. You need two devices.

**Reply by Apfelsafft** [reply]
Thanks for the answer - it make me thinking :)
The question wasn't stated clearly. I don't want to have to streams in parallel but want to have one streaming server where I can tune into FM stations or DAB+ stations. I want to switch between them, not have running two streams in parallel.

I mean it seems to obvious but I ask anyway. 
A stick can only be tuned to station - that is the reason for two?
Or is it a "frequency thing"? Could one stick stream two stations from the same DAB channel in parallel?

**Reply by AlbrechtL** [reply, marked as answer]
@Apfelsafft To listen to DAB and FM stations in parallel, you need two receivers because FM and DAB are operating on different frequencies. Here @andimik is right.

Did I understand you right that you are requesting FM support in welle.io? Technically, that is possible. Open source FM decoders can be found on different places e.g. here: https://github.com/AlbrechtL/rtl_fm_streamer. It is just the matter of doing.

As I always say: pull requests are welcome!
